src/physics/circular_filament.rs

```diff
@@ -33,49 +102,60 @@ use crate::{MU0_OVER_4PI, MU_0};
 ///   \[3\] J. Huang and J. Menard, “Development of an Auto-Convergent Free-Boundary Axisymmetric Equilibrium Solver,”
 ///         Journal of Undergraduate Research, vol. 6, Jan. 2006, Accessed: May 05, 2021. \[Online\].
 ///         Available: <https://www.osti.gov/biblio/1051805-development-auto-convergent-free-boundary-axisymmetric-equilibrium-solver>
+///
+///   \[4\] J. C. Simpson, J. E. Lane, C. D. Immer, R. C. Youngquist, and T. Steinrock,
+///         “Simple Analytic Expressions for the Magnetic Field of a Circular Current Loop,”
+///         Jan. 01, 2001. Accessed: Sep. 06, 2022. [Online]. Available: <https://ntrs.nasa.gov/citations/20010038494>
 pub fn flux_circular_filament(
-    current: &[f64],
+    ifil: &[f64],
     rfil: &[f64],
     zfil: &[f64],
     rprime: &[f64],
     zprime: &[f64],
     out_psi: &mut [f64],
 ) -> Result<(), &'static str> {
     // Check lengths; Error if they do not match
-    let m: usize = current.len();
+    let m: usize = ifil.len();
     let n: usize = rprime.len();
     if rfil.len() != m || zfil.len() != m || zprime.len() != n || out_psi.len() != n {
         return Err("Length mismatch");
     }
 
+    for i in 0..n {
+        out_psi[i] = 0.0;
+    }
+
     for i in 0..n {
         for j in 0..m {
             let rrprime = rfil[j] * rprime[i];
             let r_plus_rprime = rfil[j] + rprime[i];
             let z_minus_zprime = zfil[j] - zprime[i];
             let k2 = 4.0 * rrprime / (r_plus_rprime.powi(2) + z_minus_zprime.powi(2));
 
-            out_psi[i] += current[j]
-                * MU_0
-                * (rrprime / k2).sqrt()
-                * ((2.0 - k2) * ellipk(k2) - 2.0 * ellipe(k2));
+            out_psi[i] +=
```

Comment:
Again, it would be helpful to have an explanatory comment when the units of a variable change between different places in the code.
```suggestion
            // The contributions added here have units of ampere meter,
            // but the sum will have units of weber after it is multiplied by mu_0 below.
            out_psi[i] +=
```